core/src/main/java/org/eclipse/hono/util/HonoProtonHelper.java

```diff
@@ -164,4 +165,44 @@ public static void setDefaultCloseHandler(final ProtonSession session) {
         }
         return result;
     }
+
+    /**
+     * Close the passed link and later call its {@link ProtonLink#free()} method in an attached closeHandler.
+     * <p>
+     * Additionally supervise if this handler is called in the number of {@link Constants#DEFAULT_FREE_LINK_AFTER_CLOSE_INTERVAL_MILLIS}
+     * milliseconds. If not, the timer is triggered and the link will be freed inside the timer code.
+     *
+     * @param context The context to run the timer code on.
+     * @param link The link to close and free.
+     * @param closeHandler The handler to invoke after the link was freed.
+     */
+    public static void closeAndFree(final Context context, final ProtonLink link, final Future<Void> closeHandler) {
```

Comment:
I would appreciate an overloaded version that accepts the timeout in milliseconds as a parameter. FMPOV we should then also move the default timeout from Constants to this class since it will not need to be used anywhere else.